File: 2015/Days/02.cs
using _2015.Input;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace _2015.Days
{
    class _02
    {
        public void Run()
        {
            var dimensions = Helper.GetAsArray("02.txt").Select(o =>
            {
                var x = o.Split('x').Select(int.Parse).ToArray();
                return new
                {
                    length = x[0],
                    width = x[1],
                    height = x[2],
                    values = x.OrderBy(o => o).ToList(),
                };
            });

            var squareFeet = dimensions.Aggregate(0, (acc, curr) =>
            {
                var x = new List<int>
                {
                    curr.length * curr.width,
                    curr.width * curr.height,
                    curr.height * curr.length
                };

                return acc + x.Min() + x.Sum((val) => val * 2);
            });

            Console.WriteLine(squareFeet);

            var feetOfRibbon = dimensions.Aggregate(0, (acc, curr) =>
            {
                var ribbon = curr.values[0] * 2 + curr.values[1] * 2;
                return acc + ribbon + curr.values.Aggregate(1, (x, y) => x * y);
            });

            Console.WriteLine(feetOfRibbon);
        }
    }
}

File: 2015/days/02.ts
import { getAsArray } from '../input';
import { success } from '../utils/logger';

export default async () => {
  const dimensions = getAsArray('02.txt').map((o) => {
    const [l, w, h] = o.split('x').map(Number);
    return {
      l: l!,
      w: w!,
      h: h!,
    };
  });

  const sumArea = dimensions.reduce((acc, curr) => {
    const { l, w, h } = curr;
    const areas = [l * w, l * h, w * h];
    const sumArea = areas.reduce((a, c) => a + c * 2, 0);
    return acc + sumArea + Math.min(...areas);
  }, 0);

  success(`Part 1: ${sumArea}`);

  const sumRibbon = dimensions.reduce((acc, curr) => {
    const { l, w, h } = curr;
    const volume = l * w * h;
    const [n1, n2] = [l, w, h].sort((a, b) => a - b);
    return acc + volume + n1! * 2 + n2! * 2;
  }, 0);

  success(`Part 2: ${sumRibbon}`);
};
